### DISCUSSION TITLE
how to insert custom code for advertisement

### DISCUSSION THREAD

**Discussion by arullwizard** [PrivateBin/PrivateBin#1418] [General]
how to insert custom code for advertisement like adsense or anything which file to edit

**Comment by elrido** [comment]
You will want to look into creating your [custom template](https://github.com/PrivateBin/PrivateBin/wiki/Templates). Note that we do set a restricted content security policy (CSP) that aims to prevent loading code from 3rd parties, like Adsense will require you to do. You therefore will have to relax the CSP via the config.php file.

**Reply by rugk** [reply]
Also, it goes without saying, that "traditional" RTB/tracking ads strongly go against the aims of this project to provide a private and safe/secure tool for sharing content.

Considering finding some cooperate sponsors or so, which pay you directly.
